apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -6676,6 +6838,267 @@ enter_contract(Contract, S) ->
+    ),
+    State3 = lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_swap2(Dim) || _ <- lists:seq(1,5)], State) end,
+        State2,
+        lists:seq(1,50)
+    ),
+    State4 = lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_swap2(Dim) || _ <- lists:seq(1,20)], State) end,
+        State3,
+        lists:seq(1,50)
+    ),
+    State5 = lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_write(Dim) || _ <- lists:seq(1,2)], State) end,
+        State4,
+        lists:seq(1,50)
+    ),
+    State6 = lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_write(Dim) || _ <- lists:seq(1,5)], State) end,
+        State5,
+        lists:seq(1,50)
+    ),
+    lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_write(Dim) || _ <- lists:seq(1,20)], State) end,
+        State6,
+        lists:seq(1,50)
+    ),
+    ok.
+
+fate_vm_cross_protocol_store_big(Cfg) ->
+    state(aect_test_utils:new_state()),
+    Dim={5,5,5},
+    InitialState = store_rand_initial_state(Dim),
+    ForkHeights   = ?config(fork_heights, Cfg),
+    LimaHeight = maps:get(lima, ForkHeights),
+    IrisHeight = maps:get(iris, ForkHeights),
+    Acc = ?call(new_account, 1000000000000000000000000000000000000000 * aec_test_utils:min_gas_price()),
+    LimaSpec      = #{height => LimaHeight, vm_version => ?VM_FATE_SOPHIA_1, gas => 10000000000},
+    IrisSpec      = #{height => IrisHeight, vm_version => ?VM_FATE_SOPHIA_2, gas => 10000000000},
+    {ok, TesterContract} = compile_local_contract("StorageTester"),
+    Tester = ?call(create_contract_with_code, Acc, TesterContract, {}, LimaSpec),
+    TestOps = fun(Ops, State, Spec) -> store_rand_exe_oplist(Ops, State, Dim, Tester, Acc, Spec) end,
+
+    % Do a bunch of operations on Lima
+    State2 = lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_op(Dim) || _ <- lists:seq(1,5)], State, LimaSpec) end,
+        InitialState,
+        lists:seq(1,50)
+    ),
+    % Do operations on Iris
+    lists:foldl(
+        fun(_, State) -> TestOps([store_rand_random_op(Dim) || _ <- lists:seq(1,5)], State, IrisSpec) end,
+        State2,
+        lists:seq(1,50)
+    ),
+    ok.
+
+fate_vm_cross_protocol_store_multi_small(Cfg) ->
+    state(aect_test_utils:new_state()),
+    Dim={5,5,5},
+    InitialState = store_rand_initial_state(Dim),
+    ForkHeights   = ?config(fork_heights, Cfg),
+    LimaHeight = maps:get(lima, ForkHeights),
+    IrisHeight = maps:get(iris, ForkHeights),
+    Acc = ?call(new_account, 1000000000000000000000000000000000000000 * aec_test_utils:min_gas_price()),
+    LimaSpec      = #{height => LimaHeight, vm_version => ?VM_FATE_SOPHIA_1, gas => 10000000000},
+    IrisSpec      = #{height => IrisHeight, vm_version => ?VM_FATE_SOPHIA_2, gas => 10000000000},
+    {ok, TesterContract} = compile_local_contract("StorageTester"),
+
+    DoTest = fun() ->
+        Tester = ?call(create_contract_with_code, Acc, TesterContract, {}, LimaSpec),
```

Comment:
I don't think this mocks the protocol - essentially both contracts run under the FATE1 GC :shrug: